src/SourceBuild/content/eng/tools/BinaryToolKit/DetectBinaries.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.FileSystemGlobbing;
+
+namespace BinaryToolKit;
+
+public class DetectBinaries
+{
+    private const string Utf16Marker = "UTF-16";
+    private const int ChunkSize = 4096;
+
+    public async Task<IEnumerable<string>> ExecuteAsync(string targetDirectory)
+    {
+        Log.LogInformation($"Detecting binaries in '{targetDirectory}'...");
+
+        var matcher = new Matcher(StringComparison.Ordinal);
+        matcher.AddInclude("**/*");
+        matcher.AddExcludePatterns(new[]
+        {
+            "**/.dotnet/**",
+            "**/.git/**",
+            "**/git-info/**",
+            "**/prereqs/packages/**",
+            "**/.packages/**",
+            "artifacts/**",
+            "src/*/artifacts/**",
+        });
```

Comment:
I'm not sure if we want to have these hardcoded here. I see that .gitignore also lists some of these patterns: https://github.com/dotnet/dotnet/blob/main/.gitignore

Reply:
Thanks for the suggestion! I originally avoided using the .gitignore file since I was under the impression that a requirement for this tool is that it cannot use git. This is actually not the case, so I modified the code to make use of the .gitignore files in https://github.com/dotnet/installer/pull/18726/commits/26937be9361e6a9a7daef0433f8b47e93ece1347.